hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/wal/SyncFuture.java

```diff
@@ -65,102 +74,126 @@
    */
   private Throwable throwable;
 
-  private Thread t;
+
+  /*
+   * Fields below are created once at reset() and accessed without any lock. Should be ok as they
+   * are immutable for this instance of sync future until it is reset.
```

Comment:
Is it worth it doing all this handling just so we reuse SyncFuture instances? Maybe a later experiment would be trying to create a SyncFuture every time? Just a thought.